File: erlang/bob/src/bob.erl
-module(bob).

-export([response/1, test_version/0]).

is_question([]) -> false;
is_question([H|[]]) -> H =:= $?;
is_question([_|T]) -> is_question(T).

is_blank([]) -> true;
is_blank(String) -> false.

is_yell([]) -> false;
is_yell(String) -> false.


response(String) when (is_question(String)) -> "Sure.";
response(String) when (is_yell(String)) -> "Whoa, chill out!";
response(String) when (is_blank(String)) -> "Fine. Be that way!";
response(String) -> "Whatever.".

test_version() -> 2.

File: typescript/bob/bob.js
"use strict";
exports.__esModule = true;
var Bob = /** @class */ (function () {
    function Bob() {
    }
    Bob.prototype.hey = function (input) {
        switch (true) {
            case this.isYelling(input):
                return 'Whoa, chill out!';
            case this.isQuestioning(input):
                return 'Sure.';
            case this.isSilent(input):
                return 'Fine. Be that way!';
            default:
                return 'Whatever.';
        }
    };
    Bob.prototype.isQuestioning = function (input) {
        return input.endsWith('?');
    };
    Bob.prototype.isYelling = function (input) {
        return (input.match(/[A-Z]/) != null) && input == input.toUpperCase();
    };
    Bob.prototype.isSilent = function (input) {
        return input.trim().length == 0;
    };
    return Bob;
}());
exports["default"] = Bob;
